### PR TITLE
Add actions

### LINKED ISSUE #61: Add github actions
We have introduced some [best practices](https://pcic.uvic.ca/confluence/display/CSG/DevOps#DevOps-BestPractices) for our workflow usage. The workflows need to be updated to fall inline with these practices.

### PR DESCRIPTION
**Pull request by nikola-rados**
Resolves #61 

## Changes
- Add `ci` test workflow
- Add `pypi` publishing workflow
- Remove `travis` & `jenkins`
- Update `README`
- Mark "slow" tests

## Notes
The `ci` workflow uses `postgresql-plpython-9.5`  & `postgresql-9.5-postgis-2.4` packages rather than their `9.3` counterparts.  While these are not exactly the desired package versions, they are the oldest versions that are working.  I know we'd _like_ to have the versions perfectly match but I think this is probably ok. The alternative is very clumsy and difficult to maintain.

The changes to the `README` can be ignored for the most part.  I only added new status badges but my IDE continues to remove whitespace from `.md` files automatically.

### CODE REVIEW

File: .github/workflows/pypi-publish.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      run: |
+        python -m pip install --upgrade pip
+        pip install setuptools wheel twine
+    - name: Build and publish
+      env:
+        TWINE_USERNAME: ${{ secrets.pcic_at_pypi_username }}
+        TWINE_PASSWORD: ${{ secrets.pcic_at_pypi_password }}
+      run: |
+        python setup.py sdist bdist_wheel
+        twine upload --repository-url https://pypi.pacificclimate.org/ --skip-existing -u $TWINE_USERNAME -p $TWINE_PASSWORD dist/*
```
Comment (rod-glover) [inline .github/workflows/pypi-publish.yml:29]: Wow, this is really nice.

File: .github/workflows/python-ci.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      env:
+        PYCDS_SCHEMA_NAME: other
+      run: |
+        py.test -v --tb=short -k "not weather_anomal" tests
+    - name: Test with pytest (fast)
+      if: github.ref != 'refs/heads/master'
+      env:
+        PYCDS_SCHEMA_NAME: other
+      run: |
+        py.test -m "not slow" -v --tb=short -k "not weather_anomal" tests
```
Comment (rod-glover) [inline .github/workflows/python-ci.yml:38]: When, if ever, do the slow tests get run?

**Comment by nikola-rados** [comment]
> How and when do the slow tests get run?

The slow tests are run on the `master` branch.  We wanted to exclude them from running on dev branches to prevent it from bogging us down.  The current thought is any tests that run for 3+ seconds seconds are considered "slow".  We can change this if we find it is excluding to many, but that is the rule of thumb for now.

> Curiosity more than anything else: Why can't we use the old postgresql 9.3 packages specified in .travis.yml? Did they get removed recently?

It is not that they have been removed. You can still install and run the tests with those packages.  The issue is that the the older packages need access to some directory that is not available to actions for whatever reason.  I tried playing around with some permissions to get things to work but couldn't find a solution.  The alternative that we were running with in `Jenkins` was to use custom images with the directory made available.  This was becoming difficult to maintain and was hoping we could avoid the by using the slightly newer packages.

**Comment by rod-glover** [comment]
Those sound like reasonable choices to me. All good. Thanks!

**Pull request merged**